src/grt/src/Pin.h

```diff
@@ -106,6 +108,7 @@ class Pin
   std::map<int, std::vector<odb::Rect>> _boxesPerLayer;
   bool _isPort;
   bool _connectedToPad;
+  bool _isDriver;
```

Comment:
I don't think you really even need to record what pin is a driver. You have pointers to the db objects, right? So just write a predicate that sees if it is the first driver on the db net.